conan/tools/microsoft/toolchain.py

```diff
@@ -94,35 +124,19 @@ def _runtime_library(settings):
                                "MDd": "MultiThreadedDebugDLL"}.get(runtime, "")
         return runtime_library
 
-    def _write_config_toolchain(self, config_filename):
+    @property
+    def context_config_toolchain(self):
 
         def format_macro(key, value):
             return '%s=%s' % (key, value) if value is not None else key
 
-        toolchain_file = textwrap.dedent("""\
-            <?xml version="1.0" encoding="utf-8"?>
-            <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-              <ItemDefinitionGroup>
-                <ClCompile>
-                  <PreprocessorDefinitions>
-                     {};%(PreprocessorDefinitions)
-                  </PreprocessorDefinitions>
-                  <RuntimeLibrary>{}</RuntimeLibrary>
-                  <LanguageStandard>{}</LanguageStandard>{}{}
-                </ClCompile>
-                <ResourceCompile>
-                  <PreprocessorDefinitions>
-                     {};%(PreprocessorDefinitions)
-                  </PreprocessorDefinitions>
-                </ResourceCompile>
-              </ItemDefinitionGroup>
-              <PropertyGroup Label="Configuration">
-                <PlatformToolset>{}</PlatformToolset>
-              </PropertyGroup>
-            </Project>
-            """)
-        preprocessor_definitions = ";".join([format_macro(k, v)
-                                             for k, v in self.preprocessor_definitions.items()])
+        extra_flags = self._get_extra_flags()
```

Comment:
If you are not using the return thing ``_get_extra_flags()`` as an actual dictionary (to update or something, just getting the dict members), better make the method return a tuple of things.